pandas/_libs/tslibs/offsets.pyx

```diff
@@ -655,6 +717,21 @@ cdef class BaseOffset:
         return get_day_of_month(&dts, self._day_opt)
 
     def is_on_offset(self, dt: datetime) -> bool:
+        """
+        Return boolean whether a timestamp intersects with this frequency.
+
+        Parameters
+        ----------
+        dt : datetime.datetime
+            Timestamp to check intersections with frequency.
+
+        Examples
```

Comment:
Could we get an example where this is False as well. Think this might be useful.

Reply:
Good idea. Added an example which returns False